### PR TITLE
Create a Quiz Component

### PR DESCRIPTION
**Pull request by EvanZhouDev**
This is an in-progress and low-priority section.
- [x] Functionality
- [x] Styling
- [x] Extract as individual component

### CODE REVIEW

**Comment by ThatXliner** [comment]
Remember that for the extraction, we want to extract into a web component

**Comment by EvanZhouDev** [comment]
@ThatXliner Is it possible if the HTML is a stand-alone template which can be used. The CSS has been extracted already.
Reasoning:
- Extracting HTML is tedious with at least 5 things that need to be passed in as 'parameters'
- Since CSS is extracted copy+paste is actually not that bad

Example Template:
```html
<link rel="stylesheet" href="../assets/styles/quiz.css" />
<div id="quiz" class="p-10">
  <h1 class="font-bold text-5xl">Quiz</h1>
  <h1 class="text-4xl mt-2">For how many days was Saul (Paul) blinded?</h1>
  <div class="radio-toolbar">
    <form name="quiz">
      <input type="radio" id="option1" name="quizOptions" value="option1" />
      <label for="option1"
        ><span class="optionLetter A">A</span
        ><span class="quizAnswer">1</span></label
      >
      <input type="radio" id="option2" name="quizOptions" value="option2" />
      <label for="option2"
        ><span class="optionLetter B">B</span
        ><span class="quizAnswer">3</span></label
      >
      <input type="radio" id="option3" name="quizOptions" value="option3" />
      <label for="option3">
        <span class="optionLetter C">C</span>
        <span class="quizAnswer">7</span>
      </label>
      <input type="radio" id="option4" name="quizOptions" value="option4" />
      <label for="option4"
        ><span class="optionLetter D">D</span
        ><span class="quizAnswer">11</span></label
      >
    </form>
    <class id="quizOutput" class="text-2xl font-bold">
      Click an option to see if you are correct!
    </class>
  </div>
</div>
```

**Comment by ThatXliner** [comment]
Ever heard of https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot or template https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template?

**Comment by ThatXliner** [comment]
Use the `btn` classes provided by daisyUI. We want our UI to look consistent. Please explore the daisyUI site and maximize usage of their components.

**Comment by EvanZhouDev** [comment]
I don't know if this is good enough @ThatXliner but take a look.

Simply put this code (fill in the answers and "correct" metadata):
```html
<div id="quiz" class="p-10">
  <h1 class="font-bold text-5xl">Quiz</h1>
  <meta name="correct" content="2" />
  <h1 class="text-4xl mt-2">For how many days was Saul (Paul) blinded?</h1>
  <div class="radio-toolbar">
    <form name="quiz">
      <input type="radio" id="option1" name="quizOptions" value="option1" />
      <label for="option1"
        ><span class="optionLetter A">A</span
        ><span class="quizAnswer">1</span></label
      >
      <input type="radio" id="option2" name="quizOptions" value="option2" />
      <label for="option2"
        ><span class="optionLetter B">B</span
        ><span class="quizAnswer">3</span></label
      >
      <input type="radio" id="option3" name="quizOptions" value="option3" />
      <label for="option3">
        <span class="optionLetter C">C</span>
        <span class="quizAnswer">7</span>
      </label>
      <input type="radio" id="option4" name="quizOptions" value="option4" />
      <label for="option4"
        ><span class="optionLetter D">D</span
        ><span class="quizAnswer">11</span></label
      >
    </form>
    <class id="quizOutput" class="text-2xl font-bold">
      Click an option to see if you are correct!
    </class>
  </div>
</div>
<div class="spacing">&nbsp;</div>
```

And everything will work. You may need to adjust the positioning to your liking in your local CSS file.

I think we should focus our efforts on either finishing or polishing, and that this is good enough for a project like this.

File: src/acts-api.js [hunk 1]
```diff
@@ -50,6 +50,40 @@ export function switchPage(sID) {
     .then((data) => {
       // Safer than eval; also faster.
       Function(data)();
+      try {
```
Comment (ThatXliner) [inline src/acts-api.js:53]: We hav WebComponents for a reason

**Comment by ThatXliner** [comment]
> And everything will work. You may need to adjust the positioning to your liking in your local CSS file.

I'll see about that. I can take over a bit (if you want) and turn it into a proper WebComponent with a good API since I have minimal experience...

**Pull request merged**